android/capacitor/src/main/java/com/getcapacitor/plugin/Clipboard.java

```diff
@@ -48,23 +48,35 @@ public void read(PluginCall call) {
     ClipboardManager clipboard = (ClipboardManager)
         c.getSystemService(Context.CLIPBOARD_SERVICE);
 
-    CharSequence value = "";
-    if(clipboard.getPrimaryClipDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN)) {
-      Log.d(getLogTag(), "Got plaintxt");
-      ClipData.Item item = clipboard.getPrimaryClip().getItemAt(0);
-      value = item.getText();
-    } else {
-      Log.d(getLogTag(), "Not plaintext!");
-      ClipData.Item item = clipboard.getPrimaryClip().getItemAt(0);
-      value = item.coerceToText(this.getContext()).toString();
+    CharSequence value = null;
+
+    if (clipboard.hasPrimaryClip()) {
+      if(clipboard.getPrimaryClipDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN)) {
+        Log.d(getLogTag(), "Got plaintxt");
+        ClipData.Item item = clipboard.getPrimaryClip().getItemAt(0);
+        value = item.getText();
+      } else {
+        Log.d(getLogTag(), "Not plaintext!");
+        ClipData.Item item = clipboard.getPrimaryClip().getItemAt(0);
+        value = item.coerceToText(this.getContext()).toString();
+      }
     }
+
     JSObject ret = new JSObject();
     String type = "text/plain";
-    ret.put("value", value != null ? value : "");
-    if (value != null && value.toString().startsWith("data:")) {
-      type = value.toString().split(";")[0].split(":")[1];
+
+    if (value != null) {
+      ret.put("value", value);
+      if (value.toString().startsWith("data:")) {
+        type = value.toString().split(";")[0].split(":")[1];
+      }
     }
+    else {
+      ret.put("value", "");
+    }
+
     ret.put("type", type);
+
```

Comment:
this changes are not really needed, since you set the value to null the existing
```
ret.put("value", value != null ? value : "");	
    if (value != null && value.toString().startsWith("data:"))
```
covers all cases